### PR TITLE
MISC - visualización de las matriculas de un profesional

### PR DESCRIPTION
**Pull request by MarianoCampetella**
### Requerimiento
https://proyectos.andes.gob.ar/browse/MISC-203

### Funcionalidad desarrollada 
1. Se refactorizó el endpoint de profesionales. 

### UserStories llegó a completarse
- [X] Si
- [ ] No

### Requiere actualizaciones en la base de datos
- [ ] Si
- [X] No

### CODE REVIEW

**Comment by liquid36** [comment]
USUARIO: mcampetella  
BUILD NUMBER: [5584](http://condorpiedra.andes.gob.ar:85/5584/mochawesome.html)
CYPRESS RUN: [4519](https://dashboard.cypress.io/projects/xr7gft/runs/4519/specs)
TEST START: 2021-12-09T12:46:14.177Z
TOTAL: 402
SUCCESS: 378
FAIL: 23
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mcampetella  
BUILD NUMBER: [5585](http://condorpiedra.andes.gob.ar:85/5585/mochawesome.html)
CYPRESS RUN: [4520](https://dashboard.cypress.io/projects/xr7gft/runs/4520/specs)
TEST START: 2021-12-09T13:40:31.437Z
TOTAL: 402
SUCCESS: 389
FAIL: 12
SKIPPED: 1

File: core/tm/routes/profesional.ts [hunk 1]
```diff
@@ -535,16 +535,16 @@ router.get('/profesionales/documentos/:id', async (req, res, next) => {
...
     const opciones = {};
     let query;
     if (req.params.id) {
-        Profesional.findById(req.params.id, (err, data) => {
-            if (err) {
-                return next(err);
-            }
-            res.json(data);
-        });
+        try {
```
Comment (liquid36) [inline core/tm/routes/profesional.ts:542]: porque se agrega un try?

File: core/tm/routes/profesional.ts [hunk 2]
```diff
@@ -535,16 +535,16 @@ router.get('/profesionales/documentos/:id', async (req, res, next) => {
     res.json(await AndesDrive.find(req.params.id));
 });
 
-router.get('/profesionales/:id*?', Auth.authenticate(), (req, res, next) => {
+router.get('/profesionales/:id*?', Auth.authenticate(), async (req, res, next) => {
```
Comment (liquid36) [inline core/tm/routes/profesional.ts:538]: Ya que se esta modificando esta ruta, no viene de más: 
https://github.com/andes/andes/wiki/5---Gu%C3%ADa-de-desarrollo#no-usar-comodines-en-las-rutas

File: core/tm/routes/profesional.ts [hunk 3]
```diff
@@ -535,16 +535,12 @@ router.get('/profesionales/documentos/:id', async (req, res, next) => {
     res.json(await AndesDrive.find(req.params.id));
 });
 
-router.get('/profesionales/:id*?', Auth.authenticate(), (req, res, next) => {
+router.get('/profesionales/:id', Auth.authenticate(), async (req, res, next) => {
```
Comment (liquid36) [inline core/tm/routes/profesional.ts:538]: LA idea es separar en dos rutas así sacamos el if:

```
router.get('/profesionales/:id', Auth.authenticate(), async (req, res, next) {
    const profesional = await Profesional.findById(req.params.id);
    res.json(profesional);
});

router.get('/profesionales', Auth.authenticate(), async (req, res, next) {
    .....
});
```

Comment (liquid36) [reply core/tm/routes/profesional.ts:538]: Funcionalmente queda el mismo comportamiento pero el código más prolijo, separado por endpoint y con menos identación.

Comment (MarianoCampetella) [reply core/tm/routes/profesional.ts:538]: Corregido y muchas gracias por las aclaraciones!!

**Review by liquid36 — APPROVED** [review]

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5604](http://condorpiedra.andes.gob.ar:85/5604/mochawesome.html)
CYPRESS RUN: [4539](https://dashboard.cypress.io/projects/xr7gft/runs/4539/specs)
TEST START: 2021-12-13T18:45:47.013Z
TOTAL: 402
SUCCESS: 389
FAIL: 12
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5606](http://condorpiedra.andes.gob.ar:85/5606/mochawesome.html)
CYPRESS RUN: [4541](https://dashboard.cypress.io/projects/xr7gft/runs/4541/specs)
TEST START: 2021-12-13T19:11:09.526Z
TOTAL: 402
SUCCESS: 378
FAIL: 23
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5609](http://condorpiedra.andes.gob.ar:85/5609/mochawesome.html)
CYPRESS RUN: [4544](https://dashboard.cypress.io/projects/xr7gft/runs/4544/specs)
TEST START: 2021-12-14T11:41:57.630Z
TOTAL: 402
SUCCESS: 388
FAIL: 13
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5611](http://condorpiedra.andes.gob.ar:85/5611/mochawesome.html)
CYPRESS RUN: [4546](https://dashboard.cypress.io/projects/xr7gft/runs/4546/specs)
TEST START: 2021-12-14T12:09:57.411Z
TOTAL: 287
SUCCESS: 265
FAIL: 21
SKIPPED: 1
:warning::warning::warning::warning:
Hay archivos que no corrieron. Vuelva a intentar.

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5613](http://condorpiedra.andes.gob.ar:85/5613/mochawesome.html)
CYPRESS RUN: [4548](https://dashboard.cypress.io/projects/xr7gft/runs/4548/specs)
TEST START: 2021-12-14T12:43:00.680Z
TOTAL: 400
SUCCESS: 389
FAIL: 10
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5618](http://condorpiedra.andes.gob.ar:85/5618/mochawesome.html)
CYPRESS RUN: [4553](https://dashboard.cypress.io/projects/xr7gft/runs/4553/specs)
TEST START: 2021-12-14T13:48:07.475Z
TOTAL: 340
SUCCESS: 317
FAIL: 22
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5619](http://condorpiedra.andes.gob.ar:85/5619/mochawesome.html)
CYPRESS RUN: [4554](https://dashboard.cypress.io/projects/xr7gft/runs/4554/specs)
TEST START: 2021-12-14T14:05:27.629Z
TOTAL: 400
SUCCESS: 388
FAIL: 11
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5620](http://condorpiedra.andes.gob.ar:85/5620/mochawesome.html)
CYPRESS RUN: [4555](https://dashboard.cypress.io/projects/xr7gft/runs/4555/specs)
TEST START: 2021-12-14T14:20:15.586Z
TOTAL: 400
SUCCESS: 389
FAIL: 10
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5623](http://condorpiedra.andes.gob.ar:85/5623/mochawesome.html)
CYPRESS RUN: [4557](https://dashboard.cypress.io/projects/xr7gft/runs/4557/specs)
TEST START: 2021-12-14T15:00:43.666Z
TOTAL: 402
SUCCESS: 398
FAIL: 3
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5626](http://condorpiedra.andes.gob.ar:85/5626/mochawesome.html)
CYPRESS RUN: [4559](https://dashboard.cypress.io/projects/xr7gft/runs/4559/specs)
TEST START: 2021-12-14T15:25:57.405Z
TOTAL: 350
SUCCESS: 336
FAIL: 13
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5633](http://condorpiedra.andes.gob.ar:85/5633/mochawesome.html)
CYPRESS RUN: [4566](https://dashboard.cypress.io/projects/xr7gft/runs/4566/specs)
TEST START: 2021-12-14T17:56:51.170Z
TOTAL: 402
SUCCESS: 399
FAIL: 2
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: mbotta  
BUILD NUMBER: [5635](http://condorpiedra.andes.gob.ar:85/5635/mochawesome.html)
CYPRESS RUN: [4568](https://dashboard.cypress.io/projects/xr7gft/runs/4568/specs)
TEST START: 2021-12-15T11:39:48.995Z
TOTAL: 402
SUCCESS: 397
FAIL: 4
SKIPPED: 1

**Pull request merged**